openslides_backend/action/actions/motion/json_upload.py

```diff
@@ -0,0 +1,712 @@
+            },
+            **{
+                k: self._get_field_id(entry, v)
+                for k, v in {
+                    "category_id": "category_name",
+                    "block_id": "block",
+                }.items()
+                if entry.get(v)
+            },
+        }
+
+        errors: List[MotionActionErrorData] = []
+        if id_:
+            payload = {"id": id_, **payload}
+            errors = self.get_update_payload_integrity_error_message(
+                payload, meeting_id
+            )
+            for field in [MotionErrorType.TITLE, MotionErrorType.TEXT]:
+                if not payload.get(field):
+                    errors.append(
+                        {
+                            "type": field,
+                            "message": f"{field[0].capitalize() + field[1:]} is required",
+                        }
+                    )
+        else:
+            payload = {"meeting_id": meeting_id, **payload}
+            errors = self.get_create_payload_integrity_error_message(
+                payload, meeting_id
+            )
+
+        for err in errors:
+            entry = self._add_error_to_entry(entry, err)
+            messages.append("Error: " + err["message"])
+
+        return {"state": self.row_state, "messages": messages, "data": entry}
+
+    def setup_lookups(self, data: Iterable[Dict[str, Any]], meeting_id: int) -> None:
+        self.number_lookup = Lookup(
+            self.datastore,
+            "motion",
+            [(number, entry) for entry in data if (number := entry.get("number"))],
+            field="number",
+            mapped_fields=[],
+            global_and_filter=FilterOperator("meeting_id", "=", meeting_id),
+        )
+        self.block_lookup = self.get_lookup_dict(
+            "motion_block",
+            [title for entry in data if (title := entry.get("block"))],
+            "title",
+            and_filters=[FilterOperator("meeting_id", "=", meeting_id)],
+        )
+        self.category_lookup = self.get_lookup_dict(
+            "motion_category",
+            [name for entry in data if (name := entry.get("category_name"))],
+            "name",
+            ["prefix"],
+            and_filters=[FilterOperator("meeting_id", "=", meeting_id)],
+        )
+        self.username_lookup = self.get_lookup_dict(
+            "user",
+            list(
+                set(
+                    [
+                        username
+                        for entry in data
+                        if (
+                            usernames := [
+                                *entry.get("submitters_username", []),
+                                *entry.get("supporters_username", []),
+                            ]
+                        )
+                        for username in usernames
+                        if username
+                    ]
+                )
+            ),
+            "username",
+            ["meeting_user_ids"],
```

Comment:
add the field `meeting_ids` to the mapped_fields. This field is the calculated field for membership in a meeting. You don't need to look for the group-membership and consequently this lookup is everything you need concerning the submitter- and supporter-users.

Reply:
Done in json_upload

Reply:
Done in import as well